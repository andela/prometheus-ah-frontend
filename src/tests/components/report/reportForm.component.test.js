import configureMockStore from 'redux-mock-store';
import React from 'react';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import DefaultReportForm, { ReportForm } from '../../../components/report/ReportForm';

const mockStore = configureMockStore([thunk]);
const props = {
  article: {
    Tags: [],
    User: {
      email: null,
      firstname: null,
      id: 5,
      image: null,
      lastname: null,
      reset_password_hash: null
    },
    body: '<p>The story started few years back</p>',
    createdAt: '2018-10-15T11:27:58.250Z',
    description: '<p>the description</p>',
    id: 71,
    readingTime: '1 min read',
    slug: 'new-article',
    title: 'new article',
    updatedAt: '2018-10-15T11:27:58.250Z',
    userId: 5

  },
  articleSlug: 'new-article',
  fetchReportCategories: jest.fn(),
  hideReportModal: jest.fn(),
  modal: 'report',
  postReport: {
    error: {},
    success: {},
  },
  reportArticle: jest.fn()
};

const initialState = {
  modal: {
    current: null,
  },
  postReport: {
    success: {},
    error: {}
  }
};
const store = mockStore(initialState);

describe('Report form', () => {
  it('should render Article page correctly', () => {
    const wrapper = shallow(<ReportForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should call onChange event', () => {
    const wrapper = shallow(<ReportForm {...props} />);
    const instance = wrapper.instance();
    wrapper.setState({
      categoryId: 1, details: 'new report here', errors: {}, isLoading: true,
    });
    instance.onSubmit();
    expect(props.fetchReportCategories).toHaveBeenCalled();
    expect(props.reportArticle).toHaveBeenCalled();
    expect(props.hideReportModal).toHaveBeenCalled();
  });
  it('should set error in the state with invalid credential', () => {
    const wrapper = shallow(<ReportForm {...props} />);
    wrapper.setState({
      categoryId: 1, details: '', error: { details: ['The details field is required.'] }
    });
    const instance = wrapper.instance();
    instance.onSubmit();
    expect(wrapper.state().error).toBeTruthy();
  });
  it('should call onChange event', () => {
    const event = {
      target: {
        name: 'categoryId',
        value: '1'
      }
    };
    const wrapper = shallow(<ReportForm {...props} />);
    const categoryId = wrapper.find('#selectText');
    categoryId.simulate('change', event);
    wrapper.setState({ errors: { [event.target.name]: 'mock' } });
    wrapper.instance().onChange(event);
    expect(wrapper.instance().state[event.target.name])
      .toBe(event.target.value);
  });
  it('should test connect ReportForm', () => {
    const wrapper = shallow(<DefaultReportForm store={store} {...props} />);
    expect(wrapper.state()).toEqual({});
  });
});
