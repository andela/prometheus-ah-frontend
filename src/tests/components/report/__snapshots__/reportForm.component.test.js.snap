// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report form should render Article page correctly 1`] = `
<div>
  <div>
    <div
      className="modal-body"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            <div
              className="form-group"
            >
              <select
                className="form-control"
                id="selectText"
                name="categoryId"
                onChange={[Function]}
                value="1"
              >
                <option />
                <option />
              </select>
            </div>
            <div
              className="md-form mb-4"
            >
              <textarea
                className="form-control"
                id="myreport"
                name="details"
                onChange={[Function]}
                placeholder="Kindly provide us with more details"
                type="text"
                value=""
              />
            </div>
            <div
              className="col-md-12"
            >
              <button
                className="btn btn-danger"
                data-dismiss="modal"
                onClick={[Function]}
                type="submit"
              >
                Report
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
